pytorch_lightning/plugins/training_type/deepspeed.py

```diff
@@ -0,0 +1,226 @@
+            trainer.optimizers = [optimizer]
+            trainer.lr_schedulers = self.configure_scheduler(lr_scheduler)
+            trainer.convert_to_lightning_optimizers()
+        self.model = model
+
+    def configure_scheduler(self, lr_scheduler):
+        # this duplicates the defaults from init_optimizers
+        scheduler = {
+            'scheduler': lr_scheduler,
+            'name': None,  # no custom name
+            'interval': 'epoch',  # after epoch is over
+            'frequency': 1,  # every epoch/batch
+            'reduce_on_plateau': False,  # most often not ReduceLROnPlateau scheduler
+            'monitor': None,  # value to monitor for ReduceLROnPlateau
+            'strict': True,  # enforce that the monitor exists for ReduceLROnPlateau
+        }
+        return [scheduler]
+
+    @property
+    def lightning_module(self):
+        # the model may not be wrapped with DeepEngine & LightningDeepSpeedModule if calling this too early
+        module = getattr(self.model, "module", self.model)
+        return module.module if isinstance(module, LightningDeepSpeedModule) else module
+
+    @property
+    def distributed_sampler_kwargs(self):
+        distributed_sampler_kwargs = dict(num_replicas=self.world_size, rank=self.global_rank)
+        return distributed_sampler_kwargs
+
+    def init_optimizers(self, trainer: "Trainer", model: LightningModule) -> Tuple[List, List, List]:
+        # Skip initializing optimizers as DeepSpeed handles optimizers via config.
+        # User may have specified config options instead in configure_optimizers, but this is handled
+        # via `_format_config`
+        return [], [], []  # empty optimizers, schedulers and frequencies
+
+    def optimizer_step(self, optimizer: torch.optim.Optimizer, lambda_closure: Callable, **kwargs):
+        self.model.step(**kwargs)
+
+    def _format_config(self):
+        if not self.config:
+            raise MisconfigurationException(
+                "To use DeepSpeed you must pass in a DeepSpeed config dictionary, or path to a json config."
+                "todo: Doc Link."
+            )
+        self._format_optimizer_config()
+        self._format_batch_size_grad_accum_config()
+        self._format_precision_config()
+
+    def _format_optimizer_config(self):
+        if ("optimizer" not in self.config) or ("scheduler" not in self.config):
+            self.optimizer, self.scheduler = self.model.configure_optimizers()
+
+            if not (isinstance(self.optimizer, dict) or isinstance(self.scheduler, dict)):
+                raise MisconfigurationException(
+                    "If you have not specified an optimizer or scheduler within the DeepSpeed config"
+                    " then you must return a dict from `configure_optimizers` within the LightningModule."
+                    " See x for more information."
+                )
+
+            if not len(self.optimizer) == 1 or len(self.scheduler) == 1:
```

Comment:
```suggestion
            if len(self.optimizer) > 1 or len(self.scheduler) > 1:
```
I think otherwise the not may only apply to the optimizer part (not sure)
Is single optimizer, no scheduler supported?